app/src/main/java/org/thoughtcrime/securesms/database/MmsSmsDatabase.java

```diff
@@ -295,18 +295,21 @@ public Set<Long> getAllMessageIdsFromSenderInThread(long threadId, String serial
     return identifiedMessages;
   }
 
-  public long getLastOutgoingMessage(long threadId) {
+  public long getLastOutgoingTimestamp(long threadId) {
     String order = MmsSmsColumns.NORMALIZED_DATE_SENT + " DESC";
     String selection = MmsSmsColumns.THREAD_ID + " = " + threadId;
 
     // Try everything with resources so that they auto-close on end of scope
     try (Cursor cursor = queryTables(PROJECTION, selection, order, null)) {
       try (MmsSmsDatabase.Reader reader = readerFor(cursor)) {
         MessageRecord messageRecord;
+        long attempts = 0;
+        long maxAttempts = 20;
```

Comment:
Likely worth confirming with Kee that he's happy with only showing the delivery status on outgoing messages that are in the last 20.